prisma/migrations/20240708131554_add_client_notifications_table/migration.sql

```diff
@@ -0,0 +1,26 @@
+-- CreateTable
+CREATE TABLE "ClientNotifications" (
+    "id" UUID NOT NULL DEFAULT gen_random_uuid(),
+    "clientId" UUID NOT NULL,
+    "notificationId" UUID NOT NULL,
+    "taskId" UUID NOT NULL,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "deletedAt" TIMESTAMP(3),
+
+    CONSTRAINT "ClientNotifications_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateIndex
+CREATE INDEX "IX_ClientNotifications_notificationId" ON "ClientNotifications"("notificationId");
+
+-- CreateIndex
+CREATE INDEX "IX_ClientNotifications_clientId_taskId_deletedAt" ON "ClientNotifications"("clientId", "taskId", "deletedAt");
+
+-- CreateIndex
+CREATE UNIQUE INDEX "ClientNotifications_notificationId_key" ON "ClientNotifications"("notificationId");
+
+-- CreateIndex
+CREATE UNIQUE INDEX "ClientNotifications_clientId_taskId_deletedAt_key" ON "ClientNotifications"("clientId", "taskId", "deletedAt");
```

Comment:
it prevents more than one client <-> task notification from being created since we're just logging the task create notifications in our ClientNotifications table.

Another added bonus and performance improvement is for this query. Prisma extension handles filtering out soft deleted (deletedAt IS NULL) here so the index can optimize our query:
![image](https://github.com/copilot-platforms/tasks-app/assets/59971845/012fae0e-11d9-4f0c-9e70-67c5830e7920)

Reply:
if composite unique key of clientId, taskId and deletedAt is created

lets multiple rows with same clientId and taskId are possible with different deletedAt, no?

may be one unique key of clientId and taskId should be created.. if deleted at needs index, we can separately create that?